resources/assets/components/utilities/ScholarshipCard/GalleryBlockSignup.js

```diff
@@ -22,6 +29,33 @@ const GalleryBlockSignup = ({ campaignId, path }) => {
     },
   );
 
+  const [flash, authenticate] = useGate(
+    `OneClickSignupCampaignId:${campaignId}`,
+  );
+
+  const handleScholarshipCardShareClick = event => {
+    event.preventDefault();
+
+    trackAnalyticsEvent('clicked_scholarship_gallery_block_apply_now', {
+      action: 'button_clicked',
+      category: EVENT_CATEGORIES.siteAction,
+      label: 'scholarship_gallery_card',
+      context: {
+        url: path,
+        ...getPageContext(),
+        campaignId,
+      },
+    });
+
+    return isAuthenticated() ? handleSignup() : authenticate({ campaignId });
+  };
+
+  useEffect(() => {
+    if (isAuthenticated() && flash.campaignId === campaignId) {
```

Comment:
Since this logic jumps around a bit due to the auth flow, it might be nice to add a comment giving context:

```suggestion
    // If we're returning from the authentication flow with a "flashed" campaign ID, complete the signup:
    if (isAuthenticated() && flash.campaignId === campaignId) {
```